src/app/app.module.ts

```diff
@@ -29,11 +31,22 @@ import { MatIconModule, MatIconRegistry } from '@angular/material/icon';
   ],
   providers: [
     {
-    provide: APP_INITIALIZER,
-    useFactory: initializeRandomNames,
-    deps: [RandomNamesProvider],
-    multi: true
-  }],
+      provide: APP_INITIALIZER,
+      useFactory: initializeRandomNames,
+      deps: [RandomNamesProvider],
+      multi: true
+    },
+    // These EMULATOR InjectionTokens configure the app to use the local emulator suite when not in production.
+    // Remove them if you want to test against the real project FireStore
+    {
+      provide: USE_FIRESTORE_EMULATOR,
```

Comment:
so that is the black magic <img src="https://media1.giphy.com/media/B422mLGIeM16TLyIUA/giphy.gif"/>

Reply:
It is :)